ably/realtime_conn_spec_test.go

```diff
@@ -1810,3 +1811,181 @@ func TestRealtimeConn_RTN14(t *testing.T) {
 		}
 	}
 }
+func TestRealtimeConn_RTN14a(t *testing.T) {
+	t.Parallel()
+
+	{ // missing key
+		in := make(chan *proto.ProtocolMessage, 1)
+		out := make(chan *proto.ProtocolMessage, 16)
+
+		c, _ := ably.NewRealtime(ably.ClientOptions{}.
+			Token("fake:token").
+			AutoConnect(false).
+			Dial(ablytest.MessagePipe(in, out)))
+		c.Connect()
+		// Get the connection to CONNECTED.
+		tokenError := &proto.ErrorInfo{
+			StatusCode: http.StatusUnauthorized,
+			Code:       40140,
+		}
+		in <- &proto.ProtocolMessage{
+			Action:            proto.ActionError,
+			ConnectionDetails: &proto.ConnectionDetails{},
+			Error:             tokenError,
+		}
+		change := make(chan ably.ConnectionStateChange, 1)
+		c.Connection.OnAll(func(ev ably.ConnectionStateChange) {
+			change <- ev
+		})
+		var state ably.ConnectionStateChange
+		ablytest.Instantly.Recv(t, &state, change, t.Fatalf)
+		if expect, got := ably.ConnectionStateDisconnected, state.Current; expect != got {
+			t.Errorf("expected %v got %v", expect, got)
+		}
+		if expect, got := "missing key", c.Connection.ErrorReason(); !strings.Contains(got.Error(), expect) {
+			t.Errorf("expected %v to contain %q", got, expect)
+		}
+	}
+	{ // invalid key
+		in := make(chan *proto.ProtocolMessage, 1)
+		out := make(chan *proto.ProtocolMessage, 16)
+
+		c, _ := ably.NewRealtime(ably.ClientOptions{}.
+			Token("fake:token").
```

Comment:
If we remove this, then `c` will be nil because we call `NewREST` inside `NewRealtime` which will fail with `invalid key` error. I am testing for the `Realtime` and not `REST`

Reply:
That's a good point, yet I'd say this test relies on a bug. Both REST and Realtime should validate _all_ the ClientOptions ([RSC1b](https://docs.ably.io/client-lib-development-guide/features/#RSC1b)). The key should be validated on `NewRealtime` even if a token is passed too.

We'll get to fix that in the future, but for now, the only way to conciliate RSC1b and RTN14a is that RTN14a is about _well-formed_ (so it passes ClientOptions validation) and yet _invalid_ (non-existent, for example?) keys.